sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -0,0 +1,265 @@
+
+    private final ClientLogger logger = new ClientLogger(EncryptedBlobClientBuilder.class);
+
+    private String containerName;
+    private String blobName;
+    private String snapshot;
+
+    private BlobEncryptionPolicy encryptionPolicy;
+    private BlobDecryptionPolicy decryptionPolicy;
+
+    /**
+     * Creates a new instance of the EncryptedBlobClientBuilder
+     */
+    public EncryptedBlobClientBuilder() {
+    }
+
+    private AzureBlobStorageImpl constructImpl() {
+        Objects.requireNonNull(containerName);
+        Objects.requireNonNull(blobName);
+        Objects.requireNonNull(encryptionPolicy);
+        Objects.requireNonNull(decryptionPolicy);
+
+        HttpPipeline pipeline = super.getPipeline();
+        if (pipeline == null) {
+            pipeline = buildPipeline();
+        }
+
+        return new AzureBlobStorageBuilder()
+            .url(String.format("%s/%s/%s", endpoint, containerName, blobName))
+            .pipeline(pipeline)
+            .build();
+    }
+
+    /**
+     * Creates an {@code EncryptedBlockBlobAsyncClient} from a {@code BlockBlobAsyncClient}.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.cryptography.EncryptedBlobClientBuilder.buildEncryptedBlockBlobAsyncClient#blockblobasyncclient}
+     *
+     * @param client The client to convert.
+     * @return The encrypted client.
+     */
+    public EncryptedBlockBlobAsyncClient buildEncryptedBlockBlobAsyncClient(BlockBlobAsyncClient client) {
+        Objects.requireNonNull(encryptionPolicy);
+        Objects.requireNonNull(decryptionPolicy);
+        AzureBlobStorageImpl impl = new AzureBlobStorageBuilder()
+            .url(client.getBlobUrl().toString())
+            .pipeline(addDecryptionPolicy(client.getHttpPipeline(),
+                client.getHttpPipeline().getHttpClient()))
+            .build();
+        return new EncryptedBlockBlobAsyncClient(impl, client.getSnapshotId(), encryptionPolicy);
+    }
+
+    /**
+     * Creates an {@code EncryptedBlockBlobClient} from a {@code BlockBlobClient}.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.cryptography.EncryptedBlobClientBuilder.buildEncryptedBlockBlobClient#blockblobclient}
+     *
+     * @param client The client to convert.
+     * @return The encrypted client.
+     */
+    public EncryptedBlockBlobClient buildEncryptedBlockBlobClient(BlockBlobClient client) {
+        Objects.requireNonNull(encryptionPolicy);
+        Objects.requireNonNull(decryptionPolicy);
+        AzureBlobStorageImpl impl = new AzureBlobStorageBuilder()
+            .url(client.getBlobUrl().toString())
+            .pipeline(addDecryptionPolicy(client.getHttpPipeline(),
+                client.getHttpPipeline().getHttpClient()))
+            .build();
+        return new EncryptedBlockBlobClient(
+            new EncryptedBlockBlobAsyncClient(impl, client.getSnapshotId(), encryptionPolicy));
```

Comment:
It's unfortunate that this can't call into buildAsync. I tend to refactor some stuff like this so 114-120 are in a common place in case we need to change our configurations, but I'll leave it to you if you think that's overkill.